[ci] Update NVIDIA driver to v535 and CUDA to v12.2.1

antiagainst opened the pull request:
This commits bumps NVIDIA drivers to v535.86.05 for both CUDA and Vulkan. It also updates CUDA SDK to v12.2.1.

ScottTodd committed:
Re-enable broken tests.

pzread commented on build_tools/docker/dockerfiles/nvidia.Dockerfile:
```diff
@@ -11,12 +11,12 @@
 # compatible with the host driver (usually <= host driver version).
-ARG NVIDIA_GL_DEB="libnvidia-gl-530_530.41.03-0ubuntu0.20.04.2_amd64.deb"
-ARG NVIDIA_COMPUTE_DEB="libnvidia-compute-530_530.41.03-0ubuntu0.20.04.2_amd64.deb"
-ARG NVIDIA_COMMON_DEB="libnvidia-common-530_530.41.03-0ubuntu0.20.04.2_all.deb"
```
The NVIDIA driver on host is still 530.41

```
NVIDIA-SMI 530.41.03              Driver Version: 530.41.03    CUDA Version: 12.1
```

Do we need to upgrade them?

antiagainst replied:
I think it's fine. CUDA 12.2.x requires >=525.60.13.

antiagainst replied:
Nah. It turns out we also need to update the base VM: https://github.com/openxla/iree/actions/runs/5981136086/job/16228796396?pr=14833#step:7:164 :(

@pzread could you help with that? Thanks!

pzread replied:
I'll do it early this week

antiagainst replied:
Thanks @pzread!

pzread replied:
I set up testing runners with the new NVIDIA driver. The tests can run now (but some of them are failing): https://github.com/openxla/iree/actions/runs/5981136086/job/16283843823?pr=14833

FYI, I put these lines in the PR description so CI will run with testing runners and only run `test_gpu` job to save time. We will need to remove it before submit for a complete CI run.

```
runner-env: testing
ci-exactly: build_all, test_gpu
```

pzread replied:
The nvidia driver on all runners has been upgraded to `535.86.05`

antiagainst replied:
Nice, thanks @pzread! I'll try it again now.

antiagainst committed:
[ci] Update NVIDIA driver to v535 and CUDA to v12.2.1

This commits bumps NVIDIA drivers to v535.86.05 for both CUDA
and Vulkan. It also updates CUDA SDK to v12.2.1.

antiagainst committed:
Update SHA

antiagainst committed:
Revert "Re-enable broken tests."

This reverts commit ad5638583476f6c3d6ca6c0b2a527073ceea654b.

Groverkss commented on third_party/nvidia_sdk_download/fetch_cuda_toolkit.py:
```diff
@@ -19,7 +19,7 @@
 import sys
 
-VERSION = "12.1.1"
+VERSION = "12.2.1"
```
This change is outside CI btw. This broke my build and took some time to figure out what happened. Maybe announce it outside CI changes too?
